test/gurka/test_time_tracking.cc

```diff
@@ -0,0 +1,259 @@
+                                   midgard::kSecondsPerWeek - 10 - 60 * 60 * 3, 10, 1}));
+
+    // wrap around second of week
+    ti = baldr::TimeInfo{true, 1, 22, 5, 0, 0, cache}.reverse(10, 1);
+    ASSERT_EQ(ti, (baldr::TimeInfo{true, 1, 12, midgard::kSecondsPerWeek - 5, 10, 1}));
+
+    // cross now time
+    ti = baldr::TimeInfo{true, 1, 22, midgard::kSecondsPerWeek - 1, 5, 0, cache}.reverse(10, 1);
+    ASSERT_EQ(ti, (baldr::TimeInfo{true, 1, 12, midgard::kSecondsPerWeek - 11, 5, 1}));
+
+    // dont cross now time
+    ti = baldr::TimeInfo{true, 1, 22, midgard::kSecondsPerWeek - 1, 5, 0, cache}.reverse(2, 1);
+    ASSERT_EQ(ti, (baldr::TimeInfo{true, 1, 20, midgard::kSecondsPerWeek - 3, 3}));
+  }
+}
+
+TEST(TimeTracking, routes) {
+
+  // build a very simple graph
+  const std::string ascii_map = R"(A----B----C----D
+                                                  |
+                                                  |
+                                                  |
+                                   H----G----F----E)";
+  const gurka::ways ways = {
+      {"AB", {{"highway", "motorway"}}}, {"BC", {{"highway", "motorway"}}},
+      {"CD", {{"highway", "motorway"}}}, {"DE", {{"highway", "motorway_link"}}},
+      {"EF", {{"highway", "primary"}}},  {"FG", {{"highway", "primary"}}},
+      {"GH", {{"highway", "primary"}}},
+  };
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 100);
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_time_tracking_make",
+                               {{"mjlonir.timezone", "/path/to/timezone.sqlite"}});
+
+  // pick out a start and end ll by finding the appropriate edges in the graph
+  baldr::GraphReader reader(map.config.get_child("mjolnir"));
+  auto opp_start_edge = gurka::findEdge(reader, map.nodes, "AB", "B");
+  const baldr::GraphTile* tile = nullptr;
+  const auto* node = reader.nodeinfo(std::get<3>(opp_start_edge)->endnode(), tile);
+  auto start = node->latlng(tile->header()->base_ll());
+  auto end_edge = gurka::findEdge(reader, map.nodes, "GH", "H");
+  node = reader.nodeinfo(std::get<1>(end_edge)->endnode(), tile);
+  auto end = node->latlng(tile->header()->base_ll());
+
+  // route between them with a current time
+  auto req =
+      boost::format(
+          R"({"costing":"auto","date_time":{"type":0},"locations":[{"lon":%1%,"lat":%2%},{"lon":%3%,"lat":%4%}]})") %
+      start.first % start.second % end.first % end.second;
+  valhalla::Api api;
+  tyr::actor_t actor(map.config, reader);
+  actor.route(req.str(), nullptr, &api);
+
+  // check the timings
+  std::vector<double> times;
+  for (const auto& route : api.trip().routes()) {
+    for (const auto& leg : route.legs()) {
+      for (const auto& node : leg.node()) {
+        times.push_back(node.elapsed_time());
+      }
+    }
+  }
+  std::vector<double> expected = {0, 17.1429, 34.2857, 176.789, 220.289, 244.289, 268.289};
```

Comment:
Why the big step between 34s to 176s? Is it because of a huge turncost when turning south?

Reply:
i want to say its because its right hand driving an you make a left onto a lower class road from the ramp but... what i really did here was run this exact test on master and captured the output. this was basically a regression test